shared/actions/kbfs/index.platform.desktop.js

```diff
@@ -103,6 +111,103 @@ function openInDefault(openPath: string): Promise<*> {
   return _open(openPath)
 }
 
+function* fuseStatusSaga(): SagaGenerator<any, any> {
+  const prevFuseStatus = yield select(state => state.favorite.fuseStatus)
+
+  const status = yield call(installFuseStatusRpcPromise)
+  const action = {payload: {status}, type: 'fs:fuseStatusUpdate'}
+  yield put(action)
+
+  // If our kextStarted status changed, finish KBFS install
+  if (status.kextStarted && prevFuseStatus && !prevFuseStatus.kextStarted) {
+    console.log('Installing KBFS (kextStarted changed)')
+    yield call(installKBFSSaga)
+  }
+}
+
+function sleep(d: number): Promise<*> {
+  return new Promise((resolve, reject) => {
+    setTimeout(resolve, d)
+  })
+}
+
+function* installFuseSaga(): SagaGenerator<any, any> {
+  const result: InstallResult = yield call(installInstallFuseRpcPromise)
+  const fuseResults = result && result.componentResults
+    ? result.componentResults.filter(c => c.name === 'fuse')
+    : []
+  const kextPermissionError =
+    fuseResults.length > 0 && fuseResults[0].exitCode === ExitCodeFuseKextPermissionError
+
+  if (kextPermissionError) {
+    // Add a small delay here, since on 10.13 the OS will be a little laggy
+    // when showing a kext permission error.
+    yield call(sleep, 1000)
+  }
+
+  yield put({payload: {kextPermissionError}, type: 'fs:installFuseResult'})
+
+  yield call(fuseStatusSaga)
+
+  yield put({type: 'fs:installFuseFinished'})
+}
+
+function waitForMount(attempt: number): Promise<*> {
+  return new Promise((resolve, reject) => {
+    // Read the KBFS path waiting for files to exist, which means it's mounted
+    fs.readdir(Constants.defaultKBFSPath, (err, files) => {
+      if (!err && files.length > 0) {
+        resolve(true)
+      } else if (attempt > 15) {
+        reject(new Error(`${Constants.defaultKBFSPath} is unavailable. Please try again.`))
+      } else {
+        setTimeout(() => {
+          waitForMount(attempt + 1).then(resolve, reject)
+        }, 1000)
+      }
+    })
+  })
+}
+
+function openDefaultPath(): Promise<*> {
+  return openInDefault(Constants.defaultKBFSPath)
+}
+
+function waitForMountAndOpen(): Promise<*> {
+  return waitForMount(0).then(openDefaultPath)
+}
+
+function* waitForMountAndOpenSaga(): SagaGenerator<any, any> {
```

Comment:
It waits for /keybase to exist with files in it and then opens in Finder.
After KBFS install there is a 3-5 second lag before the mount becomes available.